core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGeneData.java

```diff
@@ -257,50 +362,107 @@ static void importSuppGeneData(File suppGeneFile) throws IOException, DaoExcepti
         reader.close(); 
     }
 
-    public static void main(String[] args) throws Exception {
-		SpringUtil.initDataSource();
-        DaoGeneOptimized daoGene = DaoGeneOptimized.getInstance();
-        daoGene.deleteAllRecords();
-        if (args.length == 0) {
-            System.out.println("command line usage:  importGenes.pl <ncbi_genes.txt> <supp-genes.txt> <microrna.txt> <all_exon_loci.bed>");
-            return;
-        }
-        ProgressMonitor.setConsoleMode(true);
-        
-        File geneFile = new File(args[0]);
-        System.out.println("Reading gene data from:  " + geneFile.getAbsolutePath());
-        int numLines = FileUtil.getNumLines(geneFile);
-        System.out.println(" --> total number of lines:  " + numLines);
-        ProgressMonitor.setMaxValue(numLines);
-        ImportGeneData.importData(geneFile);
-        ConsoleUtil.showWarnings();
-        System.err.println("Done. Restart tomcat to make sure the cache is replaced with the new data.");
-        
-        if (args.length>=2) {
-            File suppGeneFile = new File(args[1]);
-            System.out.println("Reading supp. gene data from:  " + suppGeneFile.getAbsolutePath());
-            numLines = FileUtil.getNumLines(suppGeneFile);
-            System.out.println(" --> total number of lines:  " + numLines);
-            ProgressMonitor.setMaxValue(numLines);
-            ImportGeneData.importSuppGeneData(suppGeneFile);
-        }
-        
-        if (args.length>=3) {
-            File miRNAFile = new File(args[2]);
-            System.out.println("Reading miRNA data from:  " + miRNAFile.getAbsolutePath());
-            numLines = FileUtil.getNumLines(miRNAFile);
-            System.out.println(" --> total number of lines:  " + numLines);
-            ProgressMonitor.setMaxValue(numLines);
-            ImportMicroRNAIDs.importData(miRNAFile);
+	@Override
+    public void run() {
+		try {
+			SpringUtil.initDataSource();
```

Comment:
here I would add a check to see if DB tables `gene` and `gene_alias` are not empty. If not empty, then throw an error referring the user to a documentation page with information on how to empty these tables in a safe way. So add a new doc with the contents of https://github.com/thehyve/cbioportal/wiki/Updating-gene-and-gene_alias-tables to this PR